test/JDBC/expected/BABEL-5608-vu-prepare.out

```diff
@@ -0,0 +1,238 @@
+GO
+
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid=object_id('BABEL_5608_vu_prepare_t9_6');
+GO
+~~START~~
+text
+CHECK (((((remove_accents_internal(((lower(babel_5608_vu_prepare_t9_6_col1))::text COLLATE bbf_unicode_cp1_ci_ai)))::text COLLATE bbf_unicode_cp1_cs_as) ~~* 'a%'::text COLLATE bbf_unicode_cp1_cs_as) AND (((remove_accents_internal(((lower(babel_5608_vu_prepare_t9_6_col1))::text COLLATE bbf_unicode_cp1_ci_ai)))::text >= 'a'::text COLLATE bbf_unicode_cp1_ci_ai) AND ((remove_accents_internal(((lower(babel_5608_vu_prepare_t9_6_col1))::text COLLATE bbf_unicode_cp1_ci_ai)))::text < ('a'::text COLLATE bbf_unicode_cp1_ci_ai || '?'::text COLLATE bbf_unicode_cp1_ci_ai)))))
+~~END~~
+
+
+CREATE TABLE BABEL_5608_vu_prepare_t9_7(BABEL_5608_vu_prepare_t9_7_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t9_7_c1 CHECK (LOWER(BABEL_5608_vu_prepare_t9_7_col1 COLLATE Latin1_General_CI_AI) LIKE 'abc' COLLATE Latin1_General_CI_AI));
+GO
+
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid=object_id('BABEL_5608_vu_prepare_t9_7');
+GO
+~~START~~
+text
+CHECK ((((remove_accents_internal((lower((babel_5608_vu_prepare_t9_7_col1 COLLATE bbf_unicode_cp1_ci_ai)))::text))::text COLLATE bbf_unicode_cp1_cs_as) ~~* ((remove_accents_internal(('abc'::text COLLATE "default" COLLATE bbf_unicode_cp1_ci_ai)))::text COLLATE bbf_unicode_cp1_cs_as)))
+~~END~~
+
+
+CREATE TABLE BABEL_5608_vu_prepare_t9_8(BABEL_5608_vu_prepare_t9_8_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t9_8_c1 CHECK (LOWER(BABEL_5608_vu_prepare_t9_8_col1) COLLATE Latin1_General_CI_AI LIKE 'a%' COLLATE Latin1_General_CI_AI));
+GO
+
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid=object_id('BABEL_5608_vu_prepare_t9_8');
+GO
+~~START~~
+text
+CHECK ((((remove_accents_internal(((lower(babel_5608_vu_prepare_t9_8_col1))::text COLLATE bbf_unicode_cp1_ci_ai)))::text COLLATE bbf_unicode_cp1_cs_as) ~~* ((remove_accents_internal(('a%'::text COLLATE "default" COLLATE bbf_unicode_cp1_ci_ai)))::text COLLATE bbf_unicode_cp1_cs_as)))
+~~END~~
+
+
+-- CASE 10: ESCAPE WITH LIKE
+CREATE TABLE BABEL_5608_vu_prepare_t10(BABEL_5608_vu_prepare_t10_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t10_c1 CHECK (BABEL_5608_vu_prepare_t10_col1 COLLATE Latin1_General_CI_AS LIKE '15/% %' ESCAPE '/'));
+GO
+
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid=object_id('BABEL_5608_vu_prepare_t10');
+GO
+~~START~~
+text
+CHECK (((((babel_5608_vu_prepare_t10_col1)::text COLLATE bbf_unicode_cp1_ci_as) COLLATE bbf_unicode_cp1_cs_as) ~~* (like_escape('15/% %'::text COLLATE "default", '/'::text COLLATE "default") COLLATE bbf_unicode_cp1_cs_as)))
+~~END~~
+
+
+-- CASE 11: T_CoerceViaIO
+CREATE TABLE BABEL_5608_vu_prepare_t11(BABEL_5608_vu_prepare_t11_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t11_c1 CHECK (N'123' collate Latin1_General_CI_AI LIKE CAST(123 as varchar(3))));
```

Comment:
lets also have some cases which involves VIEW

Reply:
Added in latest rev